### PR TITLE
Release microsoft.sql 2021 08 01 preview

### PR DESCRIPTION
**Pull request by ericshape**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

**2021-08-01-preview has been in production. But the swagger file has not been updated to the github swagger repo. It impacts the swagger KPI. In this PR, we address the issue by new API version release.** 

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [x] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify

**This PR would address the Swagger Completeness S360 issues.** 

  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    - [x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

**I have explained each commit in the following comments**

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ericshape** [comment]
❌ XMS_EXAMPLE_NOTFOUND_ERROR | x-ms-example not found in ReplicationLinks_Delete.Url: Microsoft.Sql/preview/2021-08-01-preview/ReplicationLinks.json#L112:17
-- | --


this validation error is expected. the new replication API would be merged by end of December with the feature owner, At that time, this valid example would be added. It would not block any API contract release.

**Comment by ericshape** [comment]
All the systemData and ReadOnly property errors are expected. They have been confirmed by the feature owner.

**Comment by ericshape** [comment]
all the breaking change are expected due to this is a new API version.

**Comment by ericshape** [comment]
**commit history detail:** 
image

**Commit 1 - 4:**

1. copy paste 2021-05-01-preview content to a new folder
2. rename the new folder as 2021-08-01-preview
3. replace all the text "2021-05-01-preview" with "2021-08-01-preview"
4. update the README.md file

**Commit 5 - 9:**
5. auto-generate 2021-08-01-preview from DSmainDev
6. Add missing example files
7. address new added pipeline issues
8. fix json format issue.

**Comment by zhenglaizhang** [comment]
@msyyc FYI, another case that python track 2 failure

@fengzhou-msft looks like .net gen throws one interesting exception, could you help take a look?

@leni-msft could you help take a look at autorestschema gen failure?

**Comment by zhenglaizhang** [comment]
Hi @jianyexi does SQL team need fix `R2020 - RequiredPropertiesMissingInResourceModel error` in https://github.com/Azure/azure-rest-api-specs/pull/16968/checks?check_run_id=4427831145 ?

**Comment by ericshape** [comment]
Waiting for the ARM sign off

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/IPv6FirewallRules.json
Comment (j5lim) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/IPv6FirewallRules.json:212]: PUT is only for a single resource creation or update. Collection PUT is a RPC violation.
https://github.com/Azure/azure-resource-manager-rpc/blob/d9c0275c7ada1a65f25ec85d88b3f8fe0bf3dc1e/v1.0/resource-api-reference.md#put-resource

**Review by j5lim — CHANGES_REQUESTED** [review]

**Comment by leni-msft** [comment]
schemas generation passed after re-run

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupGet.json
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupGet.json]: changed

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupCreateOrUpdate.json
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupCreateOrUpdate.json]: changed

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupForceFailoverAllowDataLoss.json
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/examples/InstanceFailoverGroupForceFailoverAllowDataLoss.json]: changed

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/IPv6FirewallRules.json
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/IPv6FirewallRules.json:212]: waiting for the decision from the feature owner.

Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/IPv6FirewallRules.json]: to unblock the release, we remove this batch PUT API from this version in commit: c930be5c58c9b8d759b272eafe72229b30061fe6
will add it to the next version by the feature owner.

**Review by j5lim — APPROVED** [review]
Approved from ARM side.

**Comment by msyyc** [comment]
breaking change of Python is caused by other PR since it doesn't change default tag.

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/SyncGroups.json
Comment (ArcturusZhang) [inline specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/SyncGroups.json]: still missing x-ms-enum

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Could you please apply the same change in this PR to your new swaggers?
https://github.com/Azure/azure-rest-api-specs/pull/17028/files

File: specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/SyncGroups.json
Comment (ericshape) [reply specification/sql/resource-manager/Microsoft.Sql/preview/2021-08-01-preview/SyncGroups.json]: commit https://github.com/Azure/azure-rest-api-specs/pull/16968/commits/b4801d025a17bf9c90cef971c41ce58ed441acc3 to address this issue

**Review by ArcturusZhang — APPROVED** [review]

**Comment by zhenglaizhang** [comment]
.net track2 failure is due to env issue, and after configure pipeliine to use .net core 6 the generation is good https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1252558&view=logs&j=03afb3bb-7296-55ad-aa07-ceee610c73b2&t=0f320c29-1c90-5304-d2d3-c58a6bb76437

**Pull request merged**
